src/SignatureHelpProvider.php

```diff
@@ -0,0 +1,111 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer;
+
+use Microsoft\PhpParser\Node\DelimitedList\ArgumentExpressionList;
+use Microsoft\PhpParser\Node\Expression\CallExpression;
+use Microsoft\PhpParser\Node\Expression\ArgumentExpression;
+use LanguageServer\Index\ReadableIndex;
+use LanguageServer\Protocol\{
+    Range,
+    Position,
+    SignatureHelp,
+    SignatureInformation,
+    ParameterInformation
+};
+
+class SignatureHelpProvider
+{
+    /**
+     * @var DefinitionResolver
+     */
+    private $definitionResolver;
+
+    /**
+     * @var ReadableIndex
+     */
+    private $index;
+
+    /**
+     * @param DefinitionResolver $definitionResolver
+     * @param ReadableIndex      $index
+     */
+    public function __construct(DefinitionResolver $definitionResolver, ReadableIndex $index)
+    {
+        $this->definitionResolver = $definitionResolver;
+        $this->index = $index;
+    }
+
+    /**
+     * Get the short declaration for a callable (class modifiers, function keyword, etc are stripped)
+     *
+     * @param string $declaration
+     * @return string
+     */
+    protected function getShortDeclaration(string $declaration): string
+    {
+        $parts = explode('(', $declaration, 2);
+        $name = array_reverse(explode(' ', trim($parts[0])))[0];
+        return $name . '(' . $parts[1];
```

Comment:
Okay, maybe I'm was just unable to follow this code.
- you split the declaration into two parts, the part before the `(` (function name and keywords) and the rest
- you trim whitespace from the first part (function name and keywords)
- you split the first part by (function name and keywords) by spaces
- you reverse the whole array
- you take the first (initially last) element (function name)
- you concatenate the function name, `(` and the rest

I am pretty sure `end()` is faster to get the last array element than reversing the whole array and taking the first element